### PR TITLE
[IOTDB-3727] Support clear cache in new cluster

### PR DESCRIPTION
**Pull request by lyssom**
In this PR, I support `clear cache` in new cluster.

**Function definition:**

Clear the cache of chunk, chunk metadata, timeseries metadata and bloom filter to release the memory footprint.

In cluster mode, we must consider the distinction between requests to clear the cache of a single node or the entire cluster.

The syntax definition look like:
```CLEAR CACHE (ON (LOCAL | CLUSTER))?```

### CODE REVIEW

File: confignode/src/main/java/org/apache/iotdb/confignode/client/DataNodeRequestType.java [hunk 1]
```diff
@@ -34,6 +34,7 @@ public enum DataNodeRequestType {
   CREATE_FUNCTION,
   DROP_FUNCTION,
   FLUSH,
+  CLEARCACHE,
```
Comment (liuminghui233) [inline confignode/src/main/java/org/apache/iotdb/confignode/client/DataNodeRequestType.java:37]: We should put the new enum value at the end of the list.

File: integration-test/src/test/java/org/apache/iotdb/db/it/IoTDBClearCacheIT.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+import static org.junit.Assert.fail;
+
+@RunWith(IoTDBTestRunner.class)
+@Category({LocalStandaloneIT.class, ClusterIT.class, RemoteIT.class})
+public class IoTDBClearCacheIT {
```
Comment (liuminghui233) [inline integration-test/src/test/java/org/apache/iotdb/db/it/IoTDBClearCacheIT.java:43]: This test does not verify that we properly cleared the cache. You can refer to `org.apache.iotdb.db.integration.IoTDBClearCacheIT` and migrate it simply.

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/constant/StatementType.java [hunk 3]
```diff
@@ -112,6 +112,7 @@ public enum StatementType {
   TTL,
   KILL,
   FLUSH,
+  ClearCache,
```
Comment (liuminghui233) [inline server/src/main/java/org/apache/iotdb/db/mpp/plan/constant/StatementType.java:115]: This type already exists—no need to add it.

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/statement/sys/ClearCacheStatement.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+import org.slf4j.LoggerFactory;
+
+import java.util.Collections;
+import java.util.List;
+
+public class ClearCacheStatement extends Statement implements IConfigStatement {
+
+  private static final Logger logger = LoggerFactory.getLogger(ClearCacheStatement.class);
+  /** list of storage group */
+  private List<PartialPath> storageGroups;
```
Comment (liuminghui233) [inline server/src/main/java/org/apache/iotdb/db/mpp/plan/statement/sys/ClearCacheStatement.java:39]: Maybe this attribute is a typo and we don't need it.

**Review by liuminghui233 — CHANGES_REQUESTED** [review]
BTW, we also need to update the user guide.

**Comment by lyssom** [comment]
> BTW, we also need to update the user guide.

ok，I added it

File: docs/UserGuide/Maintenance-Tools/Maintenance-Command.md [hunk 5]
```diff
@@ -48,7 +48,7 @@ IoTDB> FULL MERGE
 
 ## CLEAR CACHE
 
-Clear the cache of chunk, chunk metadata and timeseries metadata to release the memory footprint.
+Clear the cache of chunk, chunk metadata and timeseries metadata to release the memory footprint. In cluster mode, we provide commands to clear a single node cache and clear the cluster cache.
```
Comment (liuminghui233) [inline docs/UserGuide/Maintenance-Tools/Maintenance-Command.md:51]: We need to add some SQL examples.

**Review by liuminghui233 — APPROVED** [review]
LGTM :-)
I will merge this PR after the user guide is complete.

File: docs/UserGuide/Maintenance-Tools/Maintenance-Command.md (hunk 5 above)
Comment (lyssom) [reply docs/UserGuide/Maintenance-Tools/Maintenance-Command.md:51]: ok，I added sql such as CLEAR CACHE ON LOCAL and CLEAR CACHE ON CLUSTER

**Pull request merged**